From 38c8201575efbffb9b0ac53d24c16f69e547a7ae Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sun, 8 Oct 2017 04:43:30 +0200
Subject: [PATCH 0/1] efi_selftest: avoid dereferencing NULL in tpl test

The task priority levels test uses two events one passes the
notification counter as context. The other passes NULL.
Both use the same notification function. So we need to check
for NULL here.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_selftest/efi_selftest_tpl.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/efi_selftest/efi_selftest_tpl.c b/lib/efi_selftest/efi_selftest_tpl.c
index 0b78ee7595..b8c0e70262 100644
--- a/lib/efi_selftest/efi_selftest_tpl.c
+++ b/lib/efi_selftest/efi_selftest_tpl.c
@@ -26,7 +26,8 @@ static void EFIAPI notify(struct efi_event *event, void *context)
 {
 	unsigned int *count = context;
 
-	++*count;
+	if (count)
+		++*count;
 }
 
 /*
-- 
2.14.1

